app/controllers/api/v2/stars_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module Api
+  module V2
+    class StarsController < RestfulController
+      skip_before_action :load_resource
+
+      def create
+        @map = Map.find(params[:id])
+        @star = Star.new(user: current_user, map: @map)
+        authorize @map, :star?
+        create_action
+
+        if @star.errors.empty?
+          render json: @map, scope: default_scope, serializer: MapSerializer, root: serializer_root
```

Comment:
Hmm, my aim has been to ensure that there is only ever one Star connecting a user and a map... in theory you could keep hitting this endpoint and create multiple right? I'm not sure what a good way to prevent that would be